include/pros/ext_adi.h

```diff
@@ -0,0 +1,641 @@
+/**
+ * \file pros/ext_adi.h
+ *
+ * Contains prototypes for interfacing with the 3-Wire Expander.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/adi.html to learn more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2020, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_EXT_ADI_H_
+#define _PROS_EXT_ADI_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "pros/adi.h"
+#include "adi.h"
+#ifndef PROS_ERR
+#define PROS_ERR (INT32_MAX)
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+#endif
+
+#ifdef __cplusplus
+namespace c {
+#endif
+
+/******************************************************************************/
+/**                         General ADI Use Functions                        **/
+/**                                                                          **/
+/**       These functions allow for interaction with any ADI port type       **/
+/******************************************************************************/
+
+/**
+ * Gets the configuration for the given ADI port.
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of ADI Ports.
+ *
+ * \param adi_port
+ *	      The ADI port number (from 1-8, 'a'-'h', 'A'-'H') for which to return
```

Comment:
```suggestion
 *        The ADI port number (from 1-8, 'a'-'h', 'A'-'H') for which to return
```

This affects all (or at least many) of the comments in this file